### ISSUE TITLE
Waku v2 JSON-RPC REST API: Implementation

### ISSUE THREAD

**Issue by jm-clius** [status-im/nim-waku#259]
# Problem
We need to implement a REST JSON RPC based API as set out in #243 and specified in https://github.com/vacp2p/specs/issues/233.

# Acceptance criteria
A JSON-RPC REST API exists with the following functionality (more functional details in #243 ):

- Complete Store API (including the pagination excluded from POC scope in #258)
- Private API, Relay API and "Foo Subscribe" API (at this point basic end-to-end functionality could be shown)
- Filter API
- Basic Admin API, Debug API

---
**UPDATE**: Subtasks added below. 
# Subtasks:

- [x] #299 Basic end-to-end functionality

- [x] #300 Private API

- [x] #301 Basic Admin API

**NOTE**: Depending on requirements more subtasks could be added after completion of the above, e.g.:

- Full support for Push Notifications on subscriptions (not merely polling, as described in #299)
- Extended Debug and Admin APIs
- Support for named parameters, etc.

**Comment by jm-clius** [comment]
blocked by https://github.com/vacp2p/specs/issues/233

**Comment by oskarth** [comment]
Would it make sense to carve out a minimal subset of the APIs we want to initially support? Things that can be achieved in a timeboxed manner and would give us the most bang for the buck / be most useful for consumers.

**Comment by jm-clius** [comment]
> Would it make sense to carve out a minimal subset of the APIs we want to initially support?

Certainly can. And I want to write the spec in a similarly phased manner. My suggested priorities:
1. PubSub functionality (i.e. bare minimum Relay API and "SubscribeFoo" API)
2. Filter API
3. Basic Debug API
_Then:_
4. Private API and adding v1-like encryption options
5. Admin API
6. "Extended" functionality (?) in all APIs <-- here we could add "nice to haves" identified during basic implementation

I'd bundle 1, 2 & 3 together in a single "basic functionality" issue (POC already introduced basic Store API). 4 and 5 could be a separate issue. 6 we can create, when and if needed.

**Comment by jm-clius** [comment]
All subtasks completed and issues closed.
